pkgs/development/libraries/openssl/default.nix

```diff
@@ -24,7 +24,13 @@ let
     inherit version;
 
     src = fetchurl {
-      url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+      url = if lib.versionOlder version "3.0" then
+        let
+          versionFixed = builtins.replaceStrings ["."] ["_"] version;
+        in
+          "https://github.com/openssl/openssl/releases/download/OpenSSL_${versionFixed}/openssl-${version}.tar.gz"
+      else
+        "https://github.com/openssl/openssl/releases/download/openssl-${version}/openssl-${version}.tar.gz";
       inherit hash;
     };
```

Comment:
Can we not use `fetchFromGitHub` here? (Bootstrapping issues?)

Reply:
I tried fetchFromGithub and ran into a dependency loop regarding curl and fetchurl.

Reply:
Makes sense. I take it `fetchzip` would fail for similar reasons?

Reply:
Yes, with the same behavior:

* replace fetchurl -> fetchzip: fetchurl is missing
* add fetchzip and fetchurl to the argument list: infinite recursion.